src/zero/once.clj

```diff
@@ -1,77 +1,61 @@
+
+(defn- authorization-header-with-bearer-token
+  "An Authorization header with a Bearer TOKEN."
+  [token]
+  {"Authorization" (str/join \space ["Bearer" token])})
+
+(defn- get-service-account-from-environment
+  "Throw or get the service account source from ZERO_DEPLOY_ENVIRONMENT."
   []
-  (when-let [out (->> ["gcloud" "auth" "print-access-token" "--format=json"]
-                      (apply util/shell!)
-                      util/do-or-nil)]
-    (let [{:strs [client_id client_secret refresh_token token_uri]}
-          (json/read-str out)]
-      (when (and client_id client_secret refresh_token token_uri)
-        (.build (doto (UserCredentials/newBuilder)
-                  (.setClientId client_id)
-                  (.setClientSecret client_secret)
-                  (.setRefreshToken refresh_token)
-                  (.setTokenServerUri (new URI token_uri))))))))
-
-;; The re-usable credentials object for token generation
-;;
-(defonce the-cached-credentials (delay (new-user-credentials)))
-
-(defn new-credentials-from-service-account
-  "Generate scoped GoogleCredentials from a service account FILE."
-  [^String file]
-  (when file
-    (let [scopes ["email" "profile" "openid"]
-          credentials (GoogleCredentials/fromStream (FileInputStream. file))]
-      (.createScoped credentials scopes))))
-
-(defn service-account-for-env
-  "Ge the path to service account for ENVIRONMENT."
-  [environment]
-  (get-in env/stuff [environment :cromwell :wfl-service-account]))
-
-;; The re-usable environment:credentials-object map for token generation and refreshing
-;;
-(defonce the-cached-credentials-from-service-account
-         (delay
-           (let [env   (zero/throw-or-environment-keyword! (System/getenv "ENVIRONMENT"))]
-             (new-credentials-from-service-account (service-account-for-env env)))))
-
-(defn get-auth-header!
-  "Return a valid auth header. Refresh and generate the access
-   token with gcloud command if invoked from command line,
-   generate the access token from service account if invoked
-   from a live server."
+  (let [environment (util/getenv "ZERO_DEPLOY_ENVIRONMENT" "debug")
+        env (zero/throw-or-environment-keyword! environment)]
+    (get-in env/stuff [env :server :service-account])))
+
+(defn- service-account-credentials
+  "Throw or return credentials for the service account SA."
+  [{:keys [file vault] :as sa}]
+  (letfn [(jsonify [edn] (json/write-str edn :escape-slash false))]
+    (-> (cond file  (io/file file)
+              vault (-> vault util/vault-secrets jsonify .getBytes)
+              :else (throw (IllegalArgumentException. (pr-str sa))))
+      io/input-stream GoogleCredentials/fromStream
+      (.createScoped ["https://www.googleapis.com/auth/cloud-platform"
+                      "https://www.googleapis.com/auth/userinfo.email"
+                      "https://www.googleapis.com/auth/userinfo.profile"]))))
```

Comment:
just a curious question, where are these coming from? Previously we were using the OAuth2 scopes: https://developers.google.com/identity/protocols/oauth2/scopes#oauth2 which includes `openid`

Reply:
I saw an error message naming those scopes. Not driven by any advance design.
Maybe openid encompases email and enough of profile to do the trick?
I'll try it.

Reply:
Sounds good, it seems the scopes changed to use `userinfo.email` from `email`